### PR TITLE
#175494070 Return Trip

### PR DESCRIPTION
**Pull request by Liliane003**
[start:175494070]
[finish:175494070]

#### What does this PR do?

- user should be able to return trip

Description of Task to be completed?

- **This is my end point**

- POST /api/trip/return_trip

How should this be manually tested?

- Open your terminal 
- Clone this repo by running git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend
- Run git checkout ft-return-trip-175494070 to navigate to my branch
- Run git pull to make sure that you have my codes in you vs code
- Run npm install for the installation of all needed packages

Screenshots
return

Questions:
Relevant Pivot tracker Id
#175494070

### CODE REVIEW

File: src/tests/users/get-users.test.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const getUserTest = () => {
 			chai
 				.request(app)
 				.get('/api/users/profile')
-				.set('Authorization', `Bearer ${expiredToken}`)
+				.set('authorization', `bearer ${expiredToken}`)
```
Comment (BenieSereine) [inline src/tests/users/get-users.test.js]: Why did you change to lowercase?

File: src/controllers/trip.controller.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable import/no-unresolved */
+/* eslint-disable require-jsdoc */
+import tripService from '../services/trip.service';
+import ResponseService from '../services/response.service';
+
+class TripController {
+	static async returnTripController(req, res) {
+		const tripRequest = req.body;
+		console.log(tripRequest);
```
Comment (tigthor) [inline src/controllers/trip.controller.js]: Remove console log

File: src/controllers/trip.controller.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		const newTripRequest = { ...tripRequest, userId: req.userData.id };
+		console.log(newTripRequest);
+		const trip = await tripService.createTrip(newTripRequest);
+		if (!trip) {
```
Comment (tigthor) [inline src/controllers/trip.controller.js]: If conditions are put in middlewares not in controller

File: src/routes/trip.route.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express';
+import TripController from '../controllers/trip.controller';
+import { validateReturnTrip } from '../validations/trip.validation';
+import protectRoute from '../middlewares/protect-route.middleware';
+
+const router = Router();
+
+router.post('/return_trip', validateReturnTrip, protectRoute, TripController.returnTripController);
```
Comment (tigthor) [inline src/routes/trip.route.js]: Before continuing to validate input you should pass from protectRoute to validateReturnTrip

File: src/tests/trip/trip.test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../server';
+import { userToVerify, userToken, tokenToVerify, returnTrip } from '../fixtures/user.fixture';
```
[the remark this answers is not in the document]
Comment (tigthor) [reply src/tests/trip/trip.test.js:4]: Work on these Hound feedbacks

File: src/validations/trip.validation.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+			startingDate: Joi.string()
+				.regex(/^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/)
+				.required()
+				.messages({
+					'string.pattern.base': 'birthdate shoul be in the format (yyyy-mm-dd) example:(2017-11-30)',
+					'any.required': 'birthdate is required',
+					'string.empty': 'birthdate is not allowed to be empty',
+				}),
+			returningDate: Joi.string()
+				.regex(/^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/)
```
Comment (tigthor) [inline src/validations/trip.validation.js]: In this file ask yourself this question. What can happen if a user enter a returning date less than starting date that would be a disaster

**Review by tigthor — CHANGES_REQUESTED** [review]
Please work on feedbacks I left on your pr

**Comment by tigthor** [comment]
@Liliane003  Work on your pr description and how can this be tested

File: package.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
     "startDev": "nodemon --exec babel-node src/server.js",
     "lint": "./node_modules/.bin/eslint src/**/*.js",
     "lint:fixIssues": "./node_modules/.bin/eslint . --fix",
-    "migrate": "sequelize db:migrate:undo:all && sequelize db:migrate && sequelize db:seed:all",
+    "migrate": "npx sequelize db:migrate:undo:all && npx sequelize db:migrate && npx sequelize db:seed:all",
```
Comment (Niyonsengaeric) [inline package.json]: why are you changing this?

**Review by BenieSereine — APPROVED** [review]

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

**Pull request merged**